modules/wallet/seed.go

```diff
@@ -326,123 +326,155 @@ func (w *Wallet) SweepSeed(seed modules.Seed) (coins, funds types.Currency, err
 	s := newSeedScanner(seed, w.log)
 	_, maxFee := w.tpool.FeeEstimation()
 	const outputSize = 350 // approx. size in bytes of an output and accompanying signature
+	const maxOutputs = 50  // approx. number of outputs that a transaction can handle
 	s.dustThreshold = maxFee.Mul64(outputSize)
 	if err = s.scan(w.cs); err != nil {
 		return
 	}
 
-	// construct a transaction that spends the outputs
-	// TODO: this may result in transactions that are too large.
-	tb := w.StartTransaction()
-	var sweptCoins, sweptFunds types.Currency // total values of swept outputs
-	for _, output := range s.siacoinOutputs {
-		// construct a siacoin input that spends the output
-		sk := generateSpendableKey(seed, output.seedIndex)
-		tb.AddSiacoinInput(types.SiacoinInput{
-			ParentID:         types.SiacoinOutputID(output.id),
-			UnlockConditions: sk.UnlockConditions,
-		})
-		// add a signature for the input
-		sweptCoins = sweptCoins.Add(output.value)
-	}
-	for _, output := range s.siafundOutputs {
-		// construct a siafund input that spends the output
-		sk := generateSpendableKey(seed, output.seedIndex)
-		tb.AddSiafundInput(types.SiafundInput{
-			ParentID:         types.SiafundOutputID(output.id),
-			UnlockConditions: sk.UnlockConditions,
-		})
-		// add a signature for the input
-		sweptFunds = sweptFunds.Add(output.value)
-	}
+	for len(s.siacoinOutputs) > 0 || len(s.siafundOutputs) > 0 {
+		// Create temporary maps and var for the current transaction
+		siacoinOutputs := make(map[types.SiacoinOutputID]scannedOutput)
+		siafundOutputs := make(map[types.SiafundOutputID]scannedOutput)
```

Comment:
I think your way works fine too, but perhaps would be cleaner with slices instead of maps, i.e.
```go
for len(s.siacoinOutputs) > 0 || len(s.siafundOutputs) > 0 {
	var siacoinOutputs, siafundOutputs []scannedOutput
	for id, output := range s.siacoinOutputs {
		if len(siacoinOutputs) > maxOutputs {
			break
		}
		siacoinOutputs = append(siacoinOutputs, output)
		delete(s.siacoinOutputs, id)
	}
	for id, output := range s.siafundOutputs {
		if len(siacoinOutputs)+len(siafundOutputs) > maxOutputs {
			break
		}
		siafundOutputs = append(siafundOutputs, output)
		delete(s.siafundOutputs, id)
	}

	// ... use siacoinOutputs and siafundOutputs
}
```